inductiva/utils/format_utils.py

```diff
@@ -40,55 +40,60 @@ def seconds_formatter(secs: float) -> str:
     return str(datetime.timedelta(seconds=round(secs)))
 
 
-def get_tabular_str(
-    rows,
-    columns,
-    default_col_space=15,
-    override_col_space=None,
-    formatters=None,
-) -> str:
-    """Converts a list of lists to a string table.
+def apply_formatters(table_data: dict, formatters: dict):
+    """Apply formatters to a list of lists of data (rows).
 
-    Temporary solution to display tables. `pandas` dependency to print tables
-    is overkill, but since we have `pandas` anyway for now, we can use it.
+    Args:
+        table_data : Dictionary of column names and lists of data.
+        formatters : Dictionary of column names and functions to
+            apply to that column's data.
     """
-    df = pd.DataFrame(rows, columns=columns)
-    formatters = formatters or {}
-
-    col_space = {col: default_col_space for col in columns}
-    col_space.update(override_col_space or {})
-
-    # replace None with np.nan so that pandas can format them as "n/a"
-    # by passing na_rep="n/a" to to_string()
-    df.fillna(np.nan, inplace=True)
-
-    return df.to_string(
-        index=False,
-        na_rep="n/a",
-        formatters=formatters,
-        col_space=col_space,
-    )
+    for column_name, formatter in formatters.items():
+        if column_name in table_data:
+            table_data[column_name] = [
+                formatter(x) for x in table_data[column_name]
+            ]
+
+    return table_data
+
+
+def get_tabular_str(tabular_data: Union[Mapping[str, Iterable[Any]],
+                                        Iterable[Iterable[Any]]],
+                    headers: list = (),
+                    formatters: dict = None) -> str:
+    """Converts the list rows to a string table.
+
+    Args:
+        tabular_data: can be a list-of-lists (or another iterable of 
+            iterables), a list of named tuples, a dictionary of
+            iterables, an iterable of dictionaries, an iterable of
+            dataclasses (Python 3.7+)
```

Comment:
I think we are trying to be to general. In my opinion, we would assume that `get_tabular_str` gets a dictionary or a Mapping and work from there. In the functions where this is used, we can convert as we need.